File: src/sortuniq/sortuniq.nim
import std/[algorithm, strutils]

when isMainModule:
  var words: seq[string] = @[]
  for rawLine in stdin.readAll().splitLines():
    let word = rawLine.strip(chars = {'\r'})
    if word.len != 0:
      words.add(word)

  words.sort(system.cmp[string])

  var current = ""
  var count = 0
  for word in words:
    if word != current:
      if current.len != 0:
        stdout.write(current & "," & $count & "\n")
      current = word
      count = 0
    inc count

  if current.len != 0:
    stdout.write(current & "," & $count & "\n")

File: src/sortuniq/sortuniq.ml
let () =
  let words = ref [] in
  (try
     while true do
       let line = input_line stdin in
       if line <> "" then
         words := line :: !words
     done
   with End_of_file -> ());
  let words = Array.of_list !words in
  Array.sort String.compare words;
  let current = ref "" in
  let count = ref 0 in
  Array.iter
    (fun word ->
      if word <> !current then begin
        if !current <> "" then
          Printf.printf "%s,%d\n" !current !count;
        current := word;
        count := 0
      end;
      incr count)
    words;
  if !current <> "" then
    Printf.printf "%s,%d\n" !current !count
